### PR TITLE
HPCC-20102 Look at refactoring componentName and itemType in XSD schema

### PR DESCRIPTION
**Pull request by kenrowland**
Refactor itemType to remove componentName as required XSD attribute.

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by kenrowland** [comment]
@rpastrana Please review

File: initfiles/componentfiles/configschema/xsd/dafilesrv.xsd [hunk 1]
```diff
@@ -10,7 +10,7 @@
             <xs:attribute name="throttleCPULimit" type="xs:nonNegativeInteger" use="optional" default="75" hpcc:tooltip="if after the initial delay, the CPU % falls below this setting, the transaction will be allowed to continue, i.e. the limit can be exceeded this way. Overrides global settting."/>
         </xs:attributeGroup>
         <xs:sequence>
-            <xs:element name="DafilesrvProcess" hpcc:class="valueSet" hpcc:docid="daFs.t">
+            <xs:element name="DafilesrvProcess" hpcc:itemTye="dafilesrv" hpcc:class="valueSet" hpcc:docid="daFs.t">
```
Comment (rpastrana) [inline initfiles/componentfiles/configschema/xsd/dafilesrv.xsd]: is this a typo?

File: initfiles/componentfiles/configschema/xsd/hardware.xsd [hunk 2]
```diff
@@ -1,92 +1,97 @@
...
-                      <xs:enumeration value="linux"/>
-                      <xs:enumeration value="solaris"/>
-                      <xs:enumeration value="W2K"/>
-                    </xs:restriction>
-                  </xs:simpleType>
-                </xs:attribute>
-              </xs:complexType>
-            </xs:element>
+        xmlns:xs="http://www.w3.org/2001/XMLSchema" elementFormDefault="qualified" attributeFormDefault="unqualified"
+        xmlns:hpcc="someuri">
```
Comment (rpastrana) [inline initfiles/componentfiles/configschema/xsd/hardware.xsd]: we need to come up w a better uri for the hpcc ns

File: initfiles/componentfiles/configschema/xsd/hardware.xsd [hunk 3]
```diff
@@ -1,92 +1,97 @@
...
+    <xs:include schemaLocation="types.xsd"/>
+    <xs:complexType name="hardware" hpcc:class="component" hpcc:category="" hpcc:displayName="Hardware Configuration">
+        <xs:sequence>
+            <xs:element name="Hardware" hpcc:docid="????"
```
Comment (rpastrana) [inline initfiles/componentfiles/configschema/xsd/hardware.xsd]: how does this docid get set? Let's get the docs team involved if we need to.

File: initfiles/componentfiles/configschema/xsd/hardware.xsd [hunk 4]
```diff
@@ -1,92 +1,97 @@
...
+                            </xs:complexType>
+                        </xs:element>
+                    </xs:sequence>
+                </xs:complexType>
 
-        <xs:key name="domainKey">
-          <xs:selector xpath="./Domain" />
-          <xs:field xpath="@name" />
-        </xs:key>
+                <xs:key name="domainKey">
```
Comment (rpastrana) [inline initfiles/componentfiles/configschema/xsd/hardware.xsd]: keep formatting changes separate

**Review by rpastrana — CHANGES_REQUESTED** [review]
@kenrowland

**Review by kenrowland — COMMENTED** [review]
@rpastrana Please see replies to your comments

File: initfiles/componentfiles/configschema/xsd/dafilesrv.xsd (hunk 1 above)
Comment (rpastrana) [reply initfiles/componentfiles/configschema/xsd/dafilesrv.xsd]: "hpcc:itemTye" isn't that missing a p?

File: initfiles/componentfiles/configschema/xsd/hardware.xsd (hunk 2 above)
Comment (rpastrana) [reply initfiles/componentfiles/configschema/xsd/hardware.xsd]: you did put one in, make sure we use the same placeholder throughout, and let's work on establishing a real one

Comment (kenrowland) [reply initfiles/componentfiles/configschema/xsd/hardware.xsd]: I meant if we added a real URL when I said if we put one in. I'd still like to address this issue across all XSDs. I opened HPCC-20214 to make sure it's not forgotten.

@rpastrana

**Review by rpastrana — APPROVED** [review]
@kenrowland

**Comment by kenrowland** [comment]
@richardkchapman Please merge

**Comment by richardkchapman** [comment]
@kenrowland Needs rebasing

**Comment by kenrowland** [comment]
@richardkchapman  rebased. Please merge, thanks

**Pull request merged**